proposals/feature-dependencies.md

```diff
@@ -158,22 +158,35 @@ To prevent non-deterministic behavior, the algorithm will sort each **round** ac
 
 The `installsAfter` property is a "soft dependency" that influences the installation order of Features that are queued to be installed.  The effective behavior of this property is the same as `dependsOn`, with the following differences:
 
 - `installsAfter` is not recursive.  
 - `installsAfter` only influences the installation order of Features that are already set to be installed after resolving the dependency tree.
 - The Feature indicated by `installsAfter` can not provide options, nor are they able to be pinned to a specific version tag or digest.  This is unchanged from the current specification.
 
-From an implementation point of view, `installsAfter` nodes may be added as a seperate set of directed edges, just as `dependsOn` nodes are added as directed edges (see **(B1)**).  Before round-based sorting **(B2)**, an orchestrating tool should remove all `installsAfter` directed edges that do not correspond with a Feature in the `worklist` that is set to be installed.  This will ensure that `installsAfter` is only considered for Features that are already set to be installed.
+From an implementation point of view, `installsAfter` nodes may be added as a seperate set of directed edges, just as `dependsOn` nodes are added as directed edges (see **(B1)**).  Before round-based installation and sorting **(B2)**, an orchestrating tool should remove all `installsAfter` directed edges that do not correspond with a Feature in the `worklist` that is set to be installed.  In each round, a Feature can then be installed if all its requirements (both `dependsOn` and `installsAfter` dependencies) have been fulfilled in previous rounds.
 
+An implemention should fail the dependency resolution step if the evaluation of the `installsAfter` property results in an inconsistent state (eg: a circular dependency).
 
-It is not recursive, and only influences the installation order of Features that are already set to be installed via a user-defined Feature.  It is not recursive, and does not influence the installation order of dependencies of the Feature that declares the `installsAfter` property.
+### overrideFeatureInstallOrder
+
+The `overrideFeatureInstallOrder` property of `devcontainer.json` is an array of Feature IDs that are to be installed in descending priority order as soon as its dependencies outlined above are installed.  
+
+This evaluation is performed after **Round Stable Sort** and may result in a different installation order within a round.
+
+This property must not influence the dependency relationship as defined by the dependency graph (see **(B1)**) and shall only be evaulated at the round-based sorting step (see **(B2)**).  Put another way, this property cannot "pull forward" a Feature until all of its dependencies (both soft and hard) have been installed.  After a Feature's dependencies have been installed in other rounds, this property should "pull forward" each Feature as early as possible (given the order of identifiers in the array).
```

Comment:
Previously this could override the order given by soft dependencies. I think it is fine to change this since the main use case is to move features like `common-utils` which do not have any dependencies to the front of the install order. (Thinking out loud.)